script.js

```diff
@@ -1,4 +1,309 @@
+  var convertInput = input.toLowerCase().replace(/\s/g, "");
+
+  //Prints error message if input is not SPS or reversed SPS
+  var myOutputValue = `Please enter 'stone', 'paper' or 'scissors' <br>Current Score<br>User:${playerScore}<br>Com:${comScore}<br>Draws:${drawScore}`;
+
+  //If userName is not set, use first input as username
+  if (userName == "") {
+    if (input == "") {
+      return `Please input something for your username`;
+    }
+
+    userName = input;
+
+    return `Hi ${userName}, Please enter game mode to start playing.`;
+  }
+
+  //If mode is not selected, prompt to choose mode
+  if (gameModeState == 0 && gameMode == "") {
+    if (
+      input != "normal" &&
+      input != "reversed" &&
+      input != "korean" &&
+      input != "computer" &&
+      input != "word" &&
+      input != "dice"
+    ) {
+      return `Please select 'dice', 'word', 'normal' ,'reversed','computer' or 'korean' mode`;
+    }
+    gameMode = input;
+    return gameModeChecker(gameMode);
+  }
+
+  //Change game mode anytime during other gameplays
+  if (gameModeState == 1 && input == "change") {
+    gameModeState = 0;
+    gameMode = "";
+    return `Change game mode! Please enter game mode to continue playing`;
+  }
+
+  if (gameMode == "dice") {
+    myOutputValue = diceGame(input);
+  }
+
+  if (gameMode == "word") {
+    return secretWord(input);
+  }
+
+  if (gameMode == "normal") {
+    //If Input is normal SPS, run normal SPS rules.
+    if (
+      convertInput == "stone" ||
+      convertInput == "paper" ||
+      convertInput == "scissors"
+    ) {
+      myOutputValue = normalGame(convertInput, translateRPS());
+    }
+  }
+  if (gameMode == "reversed") {
+    //If input includes reversed, then run reverse SPS rules
+    if (
+      convertInput == "stone" ||
+      convertInput == "paper" ||
+      convertInput == "scissors"
+    ) {
+      myOutputValue = reverseGame(convertInput, translateRPS());
+    }
+  }
+
+  if (gameMode == "korean") {
+    if (
+      convertInput == "stone" ||
+      convertInput == "paper" ||
+      convertInput == "scissors"
+    ) {
+      //Run normal game with korean rules
+      var koreanWinner = normalGame(convertInput, translateRPS());
+      return koreanWinner;
+    }
+  }
+
+  if (gameMode == "computer") {
+    //Computer mode for auto selection
+    myOutputValue = normalGame(translateRPS(), translateRPS());
+  }
+
+  //Compare win loss with function
+  var winLossCompare = winLossPerc(playerScore, comScore);
+
+  //Output message with win/loss state, score and current game mode
+  return (
+    myOutputValue +
+    `<br> ${winLossCompare} Play Again?<br>Current Game Mode: ${gameMode}`
+  );
+};
+
+// Symbol Generation
+function symbolGenerator(input) {
+  if (input == "scissors" || input == "reversedscissors") {
+    return "✂️";
+  }
+  if (input == "paper" || input == "reversedpaper") {
+    return "📜";
+  }
+  if (input == "stone" || input == "reversedstone") {
+    return "🗿";
+  }
+}
+
```

Comment:
NICE! Great decomposition, I see you are using this `symbolGenerator()` function calls in your win/draw/lose message 

An idea for further refactoring would be to combine the message and symbol so you do not have to call a function within a function, but good thinking overall